### PR TITLE
[RFR] Us09

### PR DESCRIPTION
**Pull request by ogizanagi**
Solves #16

### CODE REVIEW

File: src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        $crawler = $this->client->request('GET', $url);
+
+        $this->assertEquals(Response::HTTP_OK, $this->client->getResponse()->getStatusCode());
+
+        //Submit text is "Preview" as we're not coming from preview page. Otherwise, would be "Update".
+        $form = $crawler->selectButton('Preview')->form();
+        $this->client->enableProfiler();
+        $this->client->submit($form, array('job[title]' => $job->getTitle() . 'Updated'));
+
+        $this->assertEquals(Response::HTTP_FOUND, $this->client->getResponse()->getStatusCode());
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/JobControllerTest.php]: please always use assertSame

File: src/SensioLabs/JobBoardBundle/Tests/Controller/ApiJobControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+class ApiJobControllerTest extends JobBoardWebTestCase
+{
+    public function testRandomIncrementsApiCount()
+    {
+        $job = (new JobGenerator())->generateJob();
+        $this->em->persist($job);
+        $this->em->flush();
+
+        $router = $this->getContainer()->get('router');
+        $this->client->request('GET', $router->generate('api_job_random'));
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Tests/Controller/ApiJobControllerTest.php]: for api routes, you should not rely on the router generation. if you api is public and used by other people. when you edit the route the tests will pass but the api will be broken for users

File: src/SensioLabs/JobBoardBundle/Resources/views/Base/manage.html.twig [hunk 3]
```diff
@@ -27,125 +27,46 @@
                     </thead>
                     <tbody>
 
-                    <tr><td colspan="5"> You have no jobs. </td></tr>
+                    {% if 0 == paginator.nbResults %}
+                        <tr><td colspan="5"> You have no jobs. </td></tr>
+                    {% else %}
+                        {% for job in paginator.currentPageResults %}
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Resources/views/Base/manage.html.twig]: you can directly use else statement with `for` in twig.

**Comment by ogizanagi** [comment]
@salahm : Thanks for your explanations. Fixed.

File: src/SensioLabs/JobBoardBundle/DependencyInjection/SensioLabsJobBoardExtension.php [hunk 4]
```diff
@@ -19,7 +19,18 @@ class SensioLabsJobBoardExtension extends Extension
...
         $loader->load('services.xml');
+
+        $mailerConfig = $config['mailers']['admin_job_notif'];
+        $definition = $container->getDefinition('mailer.job_admin_notification');
+        $arguments = $definition->getArguments();
+        $arguments[3] = array(
+            'recipient_email' => $mailerConfig['recipient_email'],
+            'sender_email' => $mailerConfig['sender_email'],
+        );
+        $definition->setArguments($arguments);
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/DependencyInjection/SensioLabsJobBoardExtension.php]: you can directly use `$definition->replaceArgument($index, $argument);` http://symfony.com/doc/current/components/dependency_injection/definitions.html

Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/DependencyInjection/SensioLabsJobBoardExtension.php]: Yep, thanks :) 
I planned this change but I'm taking more time on us11 before ^^.

**Pull request merged**